covdefaults.py

```diff
@@ -84,6 +98,81 @@ def configure(self, config: CoverageConfig) -> None:
+        ):
+            matcher = re.compile(f'.*{re.escape(f"{sep}{folder}{sep}")}.*')
+            for src in source:
+                if matcher.match(src) is not None:
+                    # now we can't cover this with one omit,
+                    # climb the tree and exclude everything else
+                    parts = os.path.relpath(src, os.getcwd()).split(sep)
+                    for at in range(1, len(parts)):
+                        level = sep.join(parts[:at])
+                        child = parts[at]
+                        for entry in os.listdir(level):
+                            if entry == child:  # dont' exclude children
+                                continue
+                            pattern = f'*/{level}/{entry}'
+                            if os.path.isdir(os.path.join(level, entry)):
+                                pattern += '/*'
+                            else:
+                                if not any(entry.endswith(e) for e in suf):
+                                    continue
+                            omit.add(pattern)
+                    break
+            else:  # blank omit any such folders
+                omit.add(f'*/{folder}{"*" if is_prefix else ""}/*')
+
+        omit.difference_update(self._subtract_omit)
+        config.set_option('run:omit', sorted(omit))
+
+    def _set_source(self, config: CoverageConfig) -> None:
+        if not isinstance(config, CoverageConfig):  # pragma: no cover
+            # https://github.com/nedbat/coveragepy/issues/967
+            config = config.config
+        source = []
+        if self._installed_package:
+            for path in _get_install_paths():
+                for pkg, dest in self._installed_package.items():
+                    base = os.path.join(path, pkg)
+                    for suffix in ('', '.py'):
+                        at = f'{base}{suffix}'
+                        if os.path.exists(at):
+                            source.append(at)
+                            if dest is not None:
+                                src_path = os.path.join(dest, pkg)
+                                config.paths[pkg] = [src_path, at]
+        # set paths to map to cwd
+        # https://coverage.readthedocs.io/en/latest/config.html#paths
+        source.append(os.getcwd())
+        config.set_option('run:source', source)
+
+
+def _get_install_paths() -> Set[str]:
+    # follow what virtualenv uses
+    distribution = Distribution({'script_args': '--no-user-cfg'})
+    # disable macOS static paths for framework
+    if hasattr(sys, '_framework'):  # pragma: no cover
+        sys._framework = None  # type:ignore # pragma: no cover
+    install = distribution.get_command_obj('install', create=True)
+    if install is None:  # pragma: no cover
+        return set()  # pragma: no cover
+    install.prefix = sys.prefix  # type:ignore
+    install.finalize_options()
+    return {install.install_platlib, install.install_purelib}  # type:ignore
```

Comment:
replace this with sysconfig, this is far too fragile

Reply:
As one who wrote virtualenv, I'm telling you that sysconfig is a lot more fragile. ``virtualenv`` set paths per distutils, not per sysconfig. ``distutils`` is always present on the Ubuntu... only ensure pip is not...

Reply:
the default ubuntu installation [does not include the `python3-distutils` package](https://www.youtube.com/watch?v=QrCePKZz5Xs&t=4h57m35s)

Reply:
hmm, then virtualenv does not support it 😆